synapse/rest/client/account.py

```diff
@@ -468,6 +468,12 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
 
         msisdn = phone_number_to_msisdn(country, phone_number)
 
+        # Didn't like the sound of logging `client_secret`, but the spec says it is
+        # "A unique string generated by the client, and used to identify the validation
+        # attempt." I.e. something to facilitate deduplication. I don't think it's a
+        # sensitive secret per se.
```

Comment:
It's not so much used to deduplicate but rather to make sure the next request(s) come from the client that started the process. I'm not sure what amount of harm can be done if someone gets a hold of someone else's client secret, but my paranoid self would prefer being cautious and not logging it if we don't need it.

Reply:
Thanks, I didn't grok that it was to be sent in another request.